scripts/oslextractmeta.py

```diff
@@ -0,0 +1,453 @@
+#! /usr/bin/env python
+
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015, Hans Hoogenboom, the appleseedhq Organization
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import json
+import glob
+import os.path
+import sys
+import argparse
+try:
+    from ConfigParser import ConfigParser
+except:
+    from configparser import ConfigParser
+import datetime
+import getpass
+import subprocess
+
+
+FileTypes = {'.oso' : "openshadinglanguage"}
+
+# metadata according to the OSL specification
+_shaderTypes = ["surface", "displacement", "light", "volume", "shader"]
+_dataTypes   = ["int", "float", "point", "vector", "normal", "color", "matrix", "string", "void"]
+_shaderKeys  = ["name", "label", "type", "help", "url", "value", "page", "widget", "units"]
+# These osl parameters are not part of the official shadinglanguage but more guidelines as how to
+# make up the interface of the shader inside a 3rd party program. Not yet decided what to do with it...
+#_parmWidgets = ["number", "string", "boolean", "checkBox", "popup", "mapper", "filename", "null"]
+#_parmInteger = ["min", "max", "sensitivity", "slider"]
+#_parmFloat   = _parmInteger + ["digits"]
+#_parmSlider  = ["slidermin", "slidermax", "slidercenter", "sliderexponent"]
+#_parmKeyword = ["output"]
+
+
+#----------------------------------------------------------
+# Functions to sanitize olsinfo output
+#----------------------------------------------------------
+
+def _error(msg, crash=False):
```

Comment:
Maybe we can have two error functions, _error() and _fatalError(), instead of using a bool arg?